### PR TITLE
feat: add support for multiple range selection

### PR DESCRIPTION
**Pull request by zhm12359**
Deployed to storybook: https://googleinterns.github.io/react-d3-chart/?path=/story/linechart--overlapped

### CODE REVIEW

**Review by naiyujiang — APPROVED** [review]

File: react-d3-chart/src/components/LineChart/LineChart.stories.tsx [hunk 1]
```diff
@@ -29,8 +29,8 @@ storiesOf('LineChart', module).add('Overlapped', () => (
     data={data}
     viewMode={'overlapped'}
     margin={{ top: 50, left: 50, right: 50, bottom: 50 }}
-    selectionCallback={(selection) => {
-      console.log(selection);
+    selectionCallback={(selections) => {
+      console.log(selections);
```
Comment (J-Pai) [inline react-d3-chart/src/components/LineChart/LineChart.stories.tsx:33]: Do we want to leave the console statements here?

File: react-d3-chart/src/utils/index.ts [hunk 2]
```diff
@@ -17,44 +18,34 @@ export const downSample = (data: LineProps[], maxPoints: number) => {
...
-};
+const getDomainHelper =
+    (data: CommonProps['data'], dimension: 'x'|'y'): Domains['xDomain'] => {
+      return [
+        Math.floor(data.reduce(
+            (minSoFar, currLine) => Math.min(
+                minSoFar,
+                currLine.reduce(
+                    (minVal, currCoord) =>
+                        Math.min(minVal, currCoord[dimension]),
```
Comment (J-Pai) [inline react-d3-chart/src/utils/index.ts:36]: I've been told I'm a separate review not to use abbreviations. Up to you if you think that's unnecessary.

File: react-d3-chart/src/components/ModeSelectionContainer/index.tsx [hunk 3]
```diff
@@ -23,9 +37,33 @@ const ModeSelectionContainer: React.FC<Props> = ({
...
+  setRangeSelectionState,
+  setRangeSelections,
+  rangeSelections,
 }) => {
+
+  const addSelection = () => {
+    onConfirmSelection(selection);
+    setRangeSelectionState({
+      selection: [0, 0],
+      eventSource: 'reset'
```
Comment (J-Pai) [inline react-d3-chart/src/components/ModeSelectionContainer/index.tsx:50]: Potential for using enum here?

File: react-d3-chart/src/components/ModeSelectionContainer/index.tsx [hunk 4]
```diff
@@ -23,9 +37,33 @@ const ModeSelectionContainer: React.FC<Props> = ({
...
+    });
+  };
+
+  const removeSelection = (selection: TRangeSelection) => {
+    const newRangeSelections = rangeSelections.filter(
+      range => range[0] !== selection[0]
+      && range[1]!==selection[1]
+    );
```
Comment (J-Pai) [inline react-d3-chart/src/components/ModeSelectionContainer/index.tsx:58]: Spacing between !==

**Review by J-Pai — APPROVED** [review]

File: react-d3-chart/src/utils/index.ts (hunk 2 above)
Comment (zhm12359) [reply react-d3-chart/src/utils/index.ts:36]: This is not from Craig - I think this line here just got reformatted by my IDE - We can discuss and potentially change in a sep PR

File: react-d3-chart/src/components/LineChart/LineChart.stories.tsx (hunk 1 above)
Comment (zhm12359) [reply react-d3-chart/src/components/LineChart/LineChart.stories.tsx:33]: Yes. This is to show this callback function works.

File: react-d3-chart/src/components/ModeSelectionContainer/index.tsx (hunk 3 above)
Comment (zhm12359) [reply react-d3-chart/src/components/ModeSelectionContainer/index.tsx:50]: this field is not really used - i will mark it as optional.

**Pull request merged**